File: crates/graphql/src/schema/objects/system_status.rs
use async_graphql::*;
use serde::Serialize;

#[derive(Default, Clone, Serialize)]
pub struct SystemStatus {
    pub resume_index: i32,
    pub resume_crc32: u32,
    pub resume_elapsed: u32,
    pub resume_offset: u32,
    pub runtime: i32,
    pub topruntime: i32,
    pub dircache_size: i32,
    pub last_screen: i8,
    pub viewer_icon_count: i32,
    pub last_volume_change: i32,
}

#[Object]
impl SystemStatus {
    async fn resume_index(&self) -> i32 {
        self.resume_index
    }

    async fn resume_crc32(&self) -> u32 {
        self.resume_crc32
    }

    async fn resume_elapsed(&self) -> u32 {
        self.resume_elapsed
    }

    async fn resume_offset(&self) -> u32 {
        self.resume_offset
    }

    async fn runtime(&self) -> i32 {
        self.runtime
    }

    async fn topruntime(&self) -> i32 {
        self.topruntime
    }

    async fn dircache_size(&self) -> i32 {
        self.dircache_size
    }

    async fn last_screen(&self) -> i8 {
        self.last_screen
    }

    async fn viewer_icon_count(&self) -> i32 {
        self.viewer_icon_count
    }

    async fn last_volume_change(&self) -> i32 {
        self.last_volume_change
    }
}

File: sdk/elixir/lib/rockbox/system_status.ex
defmodule Rockbox.SystemStatus do
  @moduledoc "Aggregate runtime information about the rockboxd instance."

  @type t :: %__MODULE__{
          resume_index: integer(),
          resume_crc32: integer(),
          resume_elapsed: integer(),
          resume_offset: integer(),
          runtime: integer(),
          topruntime: integer(),
          dircache_size: integer(),
          last_screen: integer(),
          viewer_icon_count: integer(),
          last_volume_change: integer()
        }

  defstruct [
    :resume_index,
    :resume_crc32,
    :resume_elapsed,
    :resume_offset,
    :runtime,
    :topruntime,
    :dircache_size,
    :last_screen,
    :viewer_icon_count,
    :last_volume_change
  ]
end
